### PR TITLE
Add trashbin related tests

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add more tests related to trashbin.

## Related Issue

- Related issue #1488 

## How Has This Been Tested?

:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4102/4

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIUpload</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4102/11

<p>

























</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIRenameFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4102/6

<p>





































</p></details>

File: tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature [hunk 1]
```diff
@@ -60,19 +60,19 @@ Feature: files and folders exist in the trashbin after being deleted
...
     When the user creates a folder with the name "my-empty-folder" using the webUI
     And the user creates a folder with the name "my-other-empty-folder" using the webUI
     And the user deletes folder "my-empty-folder" using the webUI
     Then as "user1" folder "my-empty-folder" should exist in trash
     But as "user1" the folder with original path "my-other-empty-folder" should not exist in trash
-    And folder "my-empty-folder" should be listed in the trashbin on the webUI
-    But folder "my-other-empty-folder" should not be listed in the trashbin on the webUI
-    When the user opens trashbin folder "my-empty-folder" using the webUI
+    When the user browses to the trashbin page using the webUI
+    And folder "my-empty-folder" should be listed on the webUI
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature]:
```suggestion
    Then folder "my-empty-folder" should be listed on the webUI
```

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 2]
```diff
@@ -29,6 +31,12 @@ Given('the user has browsed to the trashbin page', function () {
     .navigateAndWaitTillLoaded()
 })
 
+When('the user browses to the trashbin page using the webUI', function () {
```
Comment (paurakhsharma) [inline tests/acceptance/stepDefinitions/filesContext.js]:
```suggestion
When('the user browses to the trashbin page', function () {
```
Here the user is not browsing to the trashbin page using the webUI but instead directly through url

Comment (paurakhsharma) [reply tests/acceptance/stepDefinitions/filesContext.js]: Please see `the user has browsed to the favorites page using the webUI` to see what I mean by that.

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 3]
```diff
@@ -109,6 +117,13 @@ When('the user renames file/folder {string} to an invalid name {string} using th
   return client.page.FilesPageElement.filesList().renameFile(fromName, toName, false)
 })
 
+When('the user renames the following files using the webUI', function (table) {
+  for (const line of table.rows()) {
+    client.page.FilesPageElement.filesList().renameFile(line[0], line[1], false)
```
Comment (paurakhsharma) [inline tests/acceptance/stepDefinitions/filesContext.js]:
```suggestion
    client.page.FilesPageElement.filesList().renameFile(line[0], line[1])
```
I think this step is expected to succed?

File: tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature [hunk 4]
```diff
@@ -10,37 +10,37 @@ Feature: files and folders exist in the trashbin after being deleted
...
     And as "user1" file "strängé filename (duplicate #2 &).txt" should exist in trash
-    And the deleted elements should be listed in the trashbin on the webUI
-    And file "lorem.txt" should be listed in the trashbin folder "simple-folder" on the webUI
+    And the deleted elements should be listed on the webUI
+    And file "lorem.txt" should be listed in the folder "simple-folder" on the webUI
 
-  @skip @yetToImplement
   Scenario: Delete a file with problematic characters and check it is in the trashbin
-    When the user renames the following file using the webUI
+    When the user renames the following files using the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature]: this is preparation, make it an API Given step or use files with difficult names that are already in the skeleton folder

File: tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature [hunk 5]
```diff
@@ -86,10 +86,11 @@ Feature: files and folders exist in the trashbin after being deleted
...
+    And the user browses to the trashbin page using the webUI
     Then as "user1" the file with original path "lorem.txt" should exist in trash
     And as "user1" the file with original path "simple-folder/lorem.txt" should exist in trash
     And as "user1" the file with original path "strängé नेपाली folder/lorem.txt" should exist in trash
-    Then the deleted elements should be listed in the trashbin on the webUI
-    And file "lorem.txt" with path "./lorem.txt" should be listed in the trashbin on the webUI
-    And file "lorem.txt" with path "simple-folder/lorem.txt" should be listed in the trashbin on the webUI
-    And file "lorem.txt" with path "strängé नेपाली folder/lorem.txt" should be listed in the trashbin on the webUI
+    And file "lorem.txt" should be listed on the webUI
+#    And file "lorem.txt" with path "./lorem.txt" should be listed in the trashbin on the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature]: opened an issue for that #1525

File: tests/acceptance/stepDefinitions/filesContext.js (hunk 3 above)
Comment (individual-it) [inline tests/acceptance/stepDefinitions/filesContext.js]: better to use the headings of the table

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 6]
```diff
@@ -229,9 +244,109 @@ Then(/there should be (\d+) files\/folders listed on the webUI/, async function
...
+ *  "href":
+ *  "originalFilename":
+ *  "originalLocation":
+ *  "deleteTimestamp":
+ *  "lastModified":
+ * },...]
+ *
+ * @param {string} user
+ */
+const getTrashBinElements = function (user) {
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/filesContext.js]: what about putting this into webdav helper or even a new trashbin helper

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 7]
```diff
@@ -229,9 +244,109 @@ Then(/there should be (\d+) files\/folders listed on the webUI/, async function
...
+            deleteTimestamp: trash['d:propstat']['d:prop']['oc:trashbin-delete-timestamp']._text,
+            lastModified: trash['d:propstat']['d:prop']['d:getlastmodified']._text
+          })
+        })
+        resolve(trashItems)
+      })
+  })
+}
+
+Then('as {string} file/folder {string} should exist in trash', function (user, element) {
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/filesContext.js]: these steps do only WEBDAV requests and check them, so lets keep them in WebDav context or even a separate context, to keep every file as short as possible

File: tests/acceptance/stepDefinitions/filesContext.js [hunk 8]
```diff
@@ -229,9 +244,109 @@ Then(/there should be (\d+) files\/folders listed on the webUI/, async function
...
+  return client
+    .page
+    .FilesPageElement
+    .filesList()
+    .navigateToFolder(folder)
+    .waitForFileVisible(file)
+})
+
+Then('the deleted elements should be listed on the webUI', function () {
+  console.log(deletedElementsTable)
```
Comment (individual-it) [inline tests/acceptance/stepDefinitions/filesContext.js]: debug output, to be deleted?

File: tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature [hunk 9]
```diff
@@ -60,19 +60,19 @@ Feature: files and folders exist in the trashbin after being deleted
     And the deleted elements should be listed in the trashbin on the webUI
     And file "lorem.txt" should be listed in the trashbin folder "simple-folder" on the webUI
 
-  @skip @yetToImplement
   Scenario: Delete an empty folder and check it is in the trashbin
     When the user creates a folder with the name "my-empty-folder" using the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature]: Should be a Given step and done with a webDAV api requests

File: tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature [hunk 10]
```diff
@@ -60,19 +60,19 @@ Feature: files and folders exist in the trashbin after being deleted
...
     And the user creates a folder with the name "my-other-empty-folder" using the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUITrashbin/trashbinFilesFolders.feature]: Should be a Given step and done with a webDAV api requests

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4119/4

<p>





</p></details>

**Pull request merged**